### ISSUE TITLE
find_closed_orbit bugs

### ISSUE THREAD

**Issue by aoeftiger** [SixTrack/pysixtrack#61]
2 Bugs in the `Line.find_closed_orbit` algorithm:

1. the `Particle` instantiation within the `find_closed_orbit` method does not know about the mass to charge ratio, as it just takes `p0c` -- for ions of `A,Z>1` this algorithm does not find the correct closed orbit as compared to MAD-X.
2. the `coord_out` creation uses the `sigma` longitudinal coordinate while the input coordinate uses `zeta`, this should be the same variable:

```python
        def _one_turn_map(coord):
            [...]
            pcl.zeta = coord[4] ### using zeta
            pcl.delta = coord[5]

            self.track(pcl)
            coord_out = np.array(
                [pcl.x, pcl.px, pcl.y, pcl.py, pcl.sigma, pcl.delta] ### using sigma
            )

            return coord_out
```

--> both are being compared in the `_CO_error` function via `_one_turn_map(coord) - coord`, so they should be the same variable.

**Comment by aoeftiger** [comment]
To fix this, PR #50 now includes the commits [d2fe384](https://github.com/SixTrack/pysixtrack/pull/50/commits/d2fe3842c880ab18cb368fbad2985cf4728954d0) and [2dd7445](https://github.com/SixTrack/pysixtrack/pull/50/commits/2dd7445c73d5372a0f658755a272e3a981cbb237) to address this for the simple `Line.find_closed_orbit` method.

I see now that there's been a recently added `closed_orbit` module by Kostas with a correspondingly introduced advanced method `Line.find_closed_orbit_and_linear_OTM`. I've added the same **kwargs fix to address different non-protonic ions there as well in [a52aeb8](https://github.com/SixTrack/pysixtrack/pull/50/commits/a52aeb8b56f204e8cc4f38ad47443f82e735f926).

**Comment by martinschwinzerl** [comment]
Thank you & @kparasch for the contribution & the testing! We are closing this due to the merge of PR #50, please (re)open in case the problem persists!
